File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";
        String[] strArr = new String[numbers.length];
        for(int i = 0; i < numbers.length; i++){
            strArr[i] = String.valueOf(numbers[i]);
        }
        Arrays.sort(strArr, (o1, o2)->(o1 + o2).compareTo(o2 + o1));
        for(int i = strArr.length-1; i >= 0; i--){
            answer += strArr[i];
        }
        if(answer.startsWith("0")) return "0";
        return answer;
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    str_numbers = list(map(str, numbers))
    str_numbers.sort(key=lambda x: x*10, reverse=True)

    answer = ''.join(str_numbers)
    
    # 모두 0이면 
    if answer[0] == '0':
        return '0'
    
    return answer
